processor.py

```diff
@@ -165,22 +165,20 @@ def init():
         i = int(input("Your choice: "))
         if i == 0:
             init_b = False
-        elif i == 1:
-            A, B = getmatrices()
-            print(A + B)
-        elif i == 2:
-            A = getmatrix()
-            B = input("Enter constant: ")
-            if B.isdigit():
-                B = int(B)
-            else:
-                B = float(B)
+        elif i in [1, 2, 3, 5, 6]:
```

Comment:
I like this approach of doing operations.
@soumitradev What do you think ?

Reply:
Yes, this is a better approach.

However, on line 169, we use `A, B = getmatrices() if (i // 4) else (getmatrix(), 0)` if `i` is in `[1, 2, 3, 5, 6]`. Since `i // 4` will return `1` for 5 and 6, we should change this ternary operation to just `A, B = (getmatrix(), 0) if (i // 4) else getmatrices()`. The logic is inverted here. 

Another thing is that comparing is faster than integer division, so I'd suggest changing it to `A, B = (getmatrix(), 0) if (i > 4) else getmatrices()`

Reply:
@bangyen Can you fix what @soumitradev just mentioned?
We are really keeping speed our utmost priority here since for matrices of larger sizes (say, above 6x6) this *could* become a bottleneck.
Thank You.

Reply:
Fixed. I also added `or i == 2` because multiplying by a constant doesn't require a second matrix.

Reply:
@bangyen There is a bug in the code you have proposed. Please fix this.

```
(base) pseudocodenerd@sharmadhavs:~$ python processor.py
1. Add Matrices
2. Multiply matrix by a constant
3. Multiply matrices
4. Transpose matrix
5. Calculate a determinant
6. Inverse matrix
0. Exit
Your choice: 4

1. Main diagonal
2. Side Diagonal
3. Vertical line
4. Horizontal line
Your choice: 1
Enter matrix size: 3 3
Enter matrix: 
1 2 3
4 5 6
7 8 9
The result is: 
Traceback (most recent call last):
  File "processor.py", line 201, in <module>
    init()
  File "processor.py", line 195, in init
    if j in options:
UnboundLocalError: local variable 'options' referenced before assignment
```